app/views/comparisons/change_in_electricity_since_last_year/_table.html.erb

```diff
@@ -1,106 +1,32 @@
-        </td>
-        <td class="text-right">
-          <% percent_change = percent_change(change_in_electricity_since_last_year.previous_year_electricity_kwh,
-                                             change_in_electricity_since_last_year.current_year_electricity_kwh) %>
-          <%= up_downify(format_unit(percent_change, :relative_percent_0dp, true, :benchmark)) %>
-        </td>
+      <%= r.with_var val: result.previous_year_electricity_co2, unit: :co2 %>
+      <%= r.with_var val: result.current_year_electricity_co2, unit: :co2 %>
+      <%= r.with_var change: true,
+                     val: percent_change(result.previous_year_electricity_co2, result.current_year_electricity_co2),
+                     unit: :relative_percent_0dp %>
 
-        <td class="text-right">
-          <%= format_unit(change_in_electricity_since_last_year.previous_year_electricity_co2, :co2, true, :benchmark) %>
-        </td>
-        <td class="text-right">
-          <%= format_unit(change_in_electricity_since_last_year.current_year_electricity_co2, :co2, true, :benchmark) %>
-        </td>
-        <td class="text-right">
-          <% percent_change = percent_change(change_in_electricity_since_last_year.previous_year_electricity_co2,
-                                             change_in_electricity_since_last_year.current_year_electricity_co2) %>
-          <%= up_downify(format_unit(percent_change, :relative_percent_0dp, true, :benchmark)) %>
-        </td>
+      <%= r.with_var val: result.previous_year_electricity_gbp, unit: :£ %>
+      <%= r.with_var val: result.current_year_electricity_gbp, unit: :£ %>
+      <%= r.with_var change: true,
+                     val: percent_change(result.previous_year_electricity_gbp, result.current_year_electricity_gbp),
+                     unit: :relative_percent_0dp %>
 
-        <td class="text-right">
-          <%= format_unit(change_in_electricity_since_last_year.previous_year_electricity_gbp, :£, true, :benchmark) %>
-        </td>
-        <td class="text-right">
-          <%= format_unit(change_in_electricity_since_last_year.current_year_electricity_gbp, :£, true, :benchmark) %>
-        </td>
-        <td class="text-right">
-          <% percent_change = percent_change(change_in_electricity_since_last_year.previous_year_electricity_gbp,
-                                             change_in_electricity_since_last_year.current_year_electricity_gbp) %>
-          <%= up_downify(format_unit(percent_change, :relative_percent_0dp, true, :benchmark)) %>
-        </td>
-        <td class="text-right">
-          <% if change_in_electricity_since_last_year.solar_type == 'synthetic' %>
-            <%= t('common.labels.yes_label') %>
-          <% else %>
-            <%= t('common.labels.no_label') %>
-          <% end %>
-        </td>
-      </tr>
+      <%= r.with_var { result.solar_type == 'synthetic' ? t('common.labels.yes_label') : t('common.labels.no_label') } %>
```

Comment:
I think we ought to use the y_n(boolean) helper here - although I appreciate it was already like this :-)

Reply:
Yes, that's better!